### PR TITLE
fix: fix useSettings bug

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Correcting a bug I introduced in useSettings

## Test Plan
Go to Uncategorized Experiments page. You should be able to 

- toggle Show Archived switch. 
- switch pages
- sort the table by any sortable column
- add or remove columns using the Customize Columns modal (accessed through Columns button in top right)
- filter by tags, state, or user

All of these changes should persist on refresh.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ClaireNeveu — COMMENTED** [review]
The way `useSettings` is written makes it hard to judge whether these changes are correct so I really have to go off of your test plan. Unfortunately I'm unfamiliar with the broad scope associated with `useSettings`. If you could have someone more familiar with it validate your test plan that is good enough for me.

**Comment by keita-determined** [comment]
In my understanding, `useSettings` has 3 main features. 
1. store the user data in localStorage, and database. Every time user change data, LS, and DB should be updated too
2. If `skipUrlEncoding` is false or not declared, user data should be in url params
3. If there are url params in the url, these data should be override the data in LS or DB temporarily, but not update data in DB

File: webui/react/src/hooks/useSettings.ts [hunk 1]
```diff
@@ -274,14 +277,14 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
           acc.querySettings[key] = undefined;
 
           // If the settings value is invalid, set to undefined.
-          const value = partialSettings[key];
```
Comment (keita-determined) [inline webui/react/src/hooks/useSettings.ts:277]: `partialSettings[key] !== null` should be `partialSettings[key] !== undefined` or `partialSettings[key]`?

Comment (EmilyBonar) [reply webui/react/src/hooks/useSettings.ts:277]: I'm doing this (`partialSettings[key] !== null`) to avoid a typescript error.

**Review by trentwatt — APPROVED** [review]
still seeing this one 

https://user-images.githubusercontent.com/28383080/191815869-79e65456-3722-4e50-b4f4-d60f44b6ef2c.mov

overall an improvement though ¯\_(ツ)_/¯

**Pull request merged**